### ISSUE TITLE
How to use multiLabel for movie genres from csv (separated by commas)

### ISSUE THREAD

**Issue by hramtsov** [tensorflow/recommenders#368]
Good afternoon
I have a csv file with movies, which has a genres column. Each film can have several genres. Genres are separated with a comma. How do I split the genre string for each movie?
And how to submit genres to the network? How can I fix MovieModel so that genres are not like text, but like multi labels?
Thanks!

```
movies_df = pd.read_csv('drive/MyDrive/db/films_2.csv',
    usecols=['tmdbId', 'genres'])
movies_df = movies_df.astype({
    'tmdbId': np.str,
    'genres': np.str,
})

dict_movies = dict(movies_df)

movies_slices = tf.data.Dataset.from_tensor_slices(dict_movies)
movies = movies_slices.map(lambda x: {
    "movie_tmdbid": x["tmdbId"],
    "movie_genres": x["genres"],
})
```

```
unique_genres_list = np.unique(np.concatenate(list(movies.map(lambda x: tf.strings.split(x['movie_genres'], sep=',')))))
```

```
class MovieModel(tf.keras.Model):
    def __init__(self):
        super().__init__()

        self.movie_embedding = tf.keras.Sequential([
            tf.keras.layers.experimental.preprocessing.StringLookup(
                vocabulary=unique_movie_tmdbids, mask_token=None),
            tf.keras.layers.Embedding(len(unique_movie_tmdbids) + 1, 32),
        ])

        ## a temporary solution. most likely, it will work worse than the categories
        max_tokens_genres = 27

        self.genres_embedding = tf.keras.Sequential([
            tf.keras.layers.experimental.preprocessing.StringLookup(
                vocabulary=unique_genres_list, mask_token=None),
            tf.keras.layers.Embedding(len(unique_genres_list) + 1, 32)
        ])

        self.genres_vectorizer = tf.keras.layers.experimental.preprocessing.TextVectorization(
            max_tokens=max_tokens_genres,
            # split=None,
        )

        self.genres_text_embedding = tf.keras.Sequential([
            self.genres_vectorizer,
            tf.keras.layers.Embedding(max_tokens_genres, 32, mask_zero=True),
            tf.keras.layers.GlobalAveragePooling1D(),
        ])

        self.genres_vectorizer.adapt(unique_genres_list)

    def call(self, features):

        return tf.concat([
            self.movie_embedding(features["movie_tmdbid"]),
            self.genres_embedding(features['movie_genres']),
            self.genres_text_embedding(features['movie_genres']),
        ], axis=1)
```

**Comment by maciejkula** [comment]
You could consider one-hot encoding the genres, and frame this as a classification problem.
